libcontainer/specconv/spec_linux.go

```diff
@@ -41,6 +42,14 @@ var (
 		flag  int
 	}
 	complexFlags map[string]func(*configs.Mount)
+	mpolModeMap  map[specs.MemoryPolicyModeType]uint
+	mpolModeFMap map[specs.MemoryPolicyFlagType]uint
+)
+
+const (
+	// maxNumaNode is the maximum NUMA node number supported.
+	// Must be large enough to cover what the kernel supports.
+	maxNumaNode = 4095
```

Comment:
has anyone checked what the kernel actually supports?

according to numa.h:

`
#if defined(__x86_64__) || defined(__i386__)
#define NUMA_NUM_NODES  128
#else
#define NUMA_NUM_NODES  2048
#endif
`

that is later used to define the size of nodemask.

Reply:
This functionality is supported by the kernel, please see the comments in here: https://github.com/opencontainers/runc/pull/4726#discussion_r2296679488
For reference, please see the definition in the kernel source:  https://github.com/torvalds/linux/blob/master/tools/include/linux/numa.h#L12